libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs

```diff
@@ -210,7 +208,7 @@ public TestFlow AssertReply(IActivity expected, [CallerMemberName] string descri
                         throw new Exception($"{description}: Type should match");
                     }
 
-                    if (expected.AsMessageActivity().Text.Trim() != reply.AsMessageActivity().Text.Trim())
+                    if (!CompareActivities(expected.AsMessageActivity(), reply.AsMessageActivity()))
```

Comment:
I'd suggest considering adding an optional parameter `IEqualityComparer<IActivity>`. If the comparer is not provided, we use the old logic. If the comparer is provided, we use the comparer.

Your calling code can pass a custom comparer that you implement, but the rest of the code does not pass it so it continues to behave like before.